tests/test_fips.py

```diff
@@ -54,8 +50,24 @@
 RUN /bin/fips-test sha256
 """
 
-
-@pytest.mark.parametrize("runner", ALL_CONTAINERS)
+_non_fips_host_skip_mark = [pytest.mark.skipif(not host_fips_enabled(), reason="The target must run in FIPS mode for the FIPS test suite")]
+CONTAINER_IMAGES = []
+CONTAINER_IMAGES_WITH_ZYPPER = []
+for param in ALL_CONTAINERS:
+    ctr, marks = container_and_marks_from_pytest_param(param)
+    if param in LTSS_BASE_FIPS_CONTAINERS and param in CONTAINERS_WITH_ZYPPER:
+        CONTAINER_IMAGES.append(param)
+        CONTAINER_IMAGES_WITH_ZYPPER.append(param)
+    elif param in LTSS_BASE_FIPS_CONTAINERS and param not in CONTAINERS_WITH_ZYPPER:
+        CONTAINER_IMAGES.append(param)
+    elif param not in LTSS_BASE_FIPS_CONTAINERS and param in CONTAINERS_WITH_ZYPPER:
+        CONTAINER_IMAGES.append(pytest.param(ctr, marks=marks + _non_fips_host_skip_mark, id=param.id))
+        CONTAINER_IMAGES_WITH_ZYPPER.append(pytest.param(ctr, marks=marks + _non_fips_host_skip_mark, id=param.id))
+    else:
+        CONTAINER_IMAGES.append(pytest.param(ctr, marks=marks + _non_fips_host_skip_mark, id=param.id))
```

Comment:
I was thinking of something like this:
```suggestion
_non_fips_host_skip_mark = [
    pytest.mark.skipif(
        not host_fips_enabled(),
        reason="The target must run in FIPS mode for the FIPS test suite",
    )
]
CONTAINER_IMAGES = []
CONTAINER_IMAGES_WITH_ZYPPER = []
for target_list, param_list in (
    (CONTAINER_IMAGES, ALL_CONTAINERS),
    (CONTAINER_IMAGES_WITH_ZYPPER, CONTAINERS_WITH_ZYPPER),
):
    for param in param_list:
        ctr, marks = container_and_marks_from_pytest_param(param)
        if param in LTSS_BASE_FIPS_CONTAINERS:
            target_list.append(param)
        else:
            target_list.append(
                pytest.param(
                    ctr, marks=marks + _non_fips_host_skip_mark, id=param.id
                )
            )
```